src/Build/BackEnd/Components/RequestBuilder/RequestBuilder.cs

```diff
@@ -747,6 +747,12 @@ private async Task BuildAndReport()
                 // Logger threw arbitrary exception
                 thrownException = ex;
             }
+            catch (OutOfMemoryException ex)
+            {
+                // This is normally treated as a critical exception, but here at the top level we want to handle it to
```

Comment:
I don't believe this is true. OOM is catchable by the regular `catch (Exception)`.

Reply:
"critical exception" is a concept specific to this repository.

https://github.com/microsoft/msbuild/blob/ba9a1d64a7abf15a8505827c00413156a3eb7f62/src/Shared/ExceptionHandling.cs#L68-L70

Reply:
I'm going to split this change to a separate pull request.

Reply:
➡️ Now moved to #5377

Reply:
Apologies! I did not see the rest of the `catch (Exception)` block (facepalm)